### PR TITLE
Protocol v2 baseline

### PR DESCRIPTION
**Pull request by nicoayala**
- [x] DataOrder v2: Data order is now part of the DataExchange contract
- [x] DataExchange v2
- [x] openzeppelin-solidity v2: The new DataExchange contract is using `openzepplin-solidity`. The WIBToken contract is using the previous version
- [x] Fix dependencies security vulnerabilities

### CODE REVIEW

File: contracts/DataExchange.sol [hunk 1]
```diff
@@ -1,569 +1,137 @@
 pragma solidity ^0.4.24;
 
-import "zeppelin-solidity/contracts/lifecycle/TokenDestructible.sol";
-import "zeppelin-solidity/contracts/lifecycle/Pausable.sol";
-import "zeppelin-solidity/contracts/math/Math.sol";
-import "zeppelin-solidity/contracts/math/SafeMath.sol";
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
```
Comment (danielfx90) [inline contracts/DataExchange.sol]: No need for this

File: contracts/DataExchange.sol [hunk 2]
```diff
@@ -1,569 +1,137 @@
...
+import "openzeppelin-solidity/contracts/token/ERC20/IERC20.sol";
```
Comment (danielfx90) [inline contracts/DataExchange.sol]: or this

File: contracts/DataExchange.sol [hunk 3]
```diff
@@ -1,569 +1,137 @@
...
-  // @dev buyerBalance Keeps track of the buyer's balance per order-seller.
-  // TODO: Is there a better way to do this?
-  mapping(
-    address => mapping(address => mapping(address => uint256))
-  ) public buyerBalance;
-
-  // @dev buyerRemainingBudgetForAudits Keeps track of the buyer's remaining
-  // budget from the initial one set on the `DataOrder`
-  mapping(address => mapping(address => uint256)) public buyerRemainingBudgetForAudits;
+  IERC20 public token;
```
Comment (danielfx90) [inline contracts/DataExchange.sol]: we don't need this anymore

File: contracts/DataExchange.sol [hunk 4]
```diff
@@ -1,569 +1,137 @@
...
    * @notice Contract constructor.
    * @param tokenAddress Address of the WIBToken token address.
-   * @param ownerAddress Address of the DataExchange owner.
    */
-  constructor(
-    address tokenAddress,
-    address ownerAddress
-  ) public validAddress(tokenAddress) validAddress(ownerAddress) {
-    require(tokenAddress != ownerAddress);
+  constructor(address tokenAddress) public {
```
Comment (danielfx90) [inline contracts/DataExchange.sol]: I would remove the constructor. We don't need it anymore.

File: contracts/DataExchange.sol [hunk 5]
```diff
@@ -1,569 +1,137 @@
...
+    require(tokenAddress != address(0));
+    token = IERC20(tokenAddress);
+  }
 
-    token = WIBToken(tokenAddress);
-    minimumInitialBudgetForAudits = 0;
-    transferOwnership(ownerAddress);
+  function isSenderNotary() private view returns (bool) {
```
Comment (manelli) [inline contracts/DataExchange.sol]: Should this `function` be a `modifier` instead?

**Pull request merged**